tests/unit-tests/shell/test_basic_accessibility_manager.cpp

```diff
@@ -245,3 +246,93 @@ TEST_F(TestBasicAccessibilityManager, calling_set_max_speed_calls_set_max_speed_
     basic_accessibility_manager.mousekeys_enabled(true);
     basic_accessibility_manager.max_speed(max_x, max_y);
 }
+
+TEST_F(TestBasicAccessibilityManager, calling_simulated_secondary_click_hold_duration_calls_the_corresponding_method_on_transformer)
+{
+    auto const expected_duration = std::chrono::milliseconds{333};
+    EXPECT_CALL(*mock_simulated_secondary_click_transformer, hold_duration(expected_duration));
+
+    basic_accessibility_manager.simulated_secondary_click_enabled(true);
+    basic_accessibility_manager.simulated_secondary_click().hold_duration(expected_duration);
+}
+
+TEST_F(TestBasicAccessibilityManager, setting_on_hold_start_sets_it_on_transformer)
+{
+    auto const expected_on_hold_start = []
+    {
+    };
+
+    EXPECT_CALL(*mock_simulated_secondary_click_transformer, on_hold_start(_));
+
+    basic_accessibility_manager.simulated_secondary_click().on_hold_start(expected_on_hold_start);
+}
```

Comment:
[Non blocking] This looks somewhat like testing implementation details, not the desired behavior.

We don't need to care that `mock_simulated_secondary_click_transformer, on_hold_start()` is called. If the interaction between `BasicAccessibilityManager` and `Transformer` were changed _without changing functionality_ then we'd need to update this test.

What we care about is that `expected_on_hold_start()` is called when the hold starts.